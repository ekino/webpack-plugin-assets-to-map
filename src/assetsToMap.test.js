const AssetsToMap = require("./assetsToMap");
const fs = require("fs");

function cleanResultFiles() {
  const files = "./src/mock/result/final";

  fs.unlink(files + ".json", err => {
    if (err) {
    }
  });
  fs.unlink(files + ".yaml", err => {
    if (err) {
    }
  });
  fs.unlink(files + ".toml", err => {
    if (err) {
    }
  });
}

function cleanSourceFilesMock() {
  fs.unlink("./src/mock/mock.js", err => {
    if (err) {
    }
  });
  fs.unlink("./src/mock/mock2.css", err => {
    if (err) {
    }
  });
}

function cpSourceFilesMock() {
  fs.copyFileSync(
    "./src/mock/sourceFiles/mock.js",
    "./src/mock/mock.js",
    err => {
      if (err) throw err;
    }
  );
  fs.copyFileSync(
    "./src/mock/sourceFiles/mock2.css",
    "./src/mock/mock2.css",
    err => {
      if (err) throw err;
    }
  );
}

describe("AssetsToMap", () => {
  beforeEach(async (done) => {
    await cpSourceFilesMock();
    done();
  });

  afterEach(async (done) => {
    await cleanResultFiles();
    await cleanSourceFilesMock();

    done();
  });

  test("hello world", () => {
    const compiler = {};
    compiler.plugin = jest.fn();
    AssetsToMap.prototype.apply(compiler);
    expect(compiler.plugin).toHaveBeenCalled();
  });

  test("writeFiles no files provided", () => {
    // with
    const plugin = new AssetsToMap();

    // when
    const result = plugin.writeFile();

    // then
    expect(result).toBeFalsy();
  });

  test("writeFiles files is not an object", () => {
    // with
    const config = {
      files: ["test.js", "test2.css"],
    };
    const plugin = new AssetsToMap();

    // when
    let result = plugin.writeFile(config);

    // then
    expect(result).toBeFalsy();
  });

  test("writeFiles sourcePath and outputPath should be okay with a trailing slash", () => {
    // with
    const config = {
      files: {
        success: "mock.js",
        style: "mock2.css",
      },
      filename: "final",
      sourcePath: "./src/mock/",
      outputPath: "./src/mock/result/",
    };
    const plugin = new AssetsToMap(config);

    // when
    let result = plugin.writeFile();

    let data = fs.readFileSync("./src/mock/result/final.json").toString();

    // then
    expect(result).toBeTruthy();
    expect(data).toMatchSnapshot();
  });

  test("writeFiles sourcePath and outputPath should be okay without a trailing slash", () => {
    // with
    const config = {
      files: {
        success: "mock.js",
        style: "mock2.css",
      },
      filename: "final",
      sourcePath: "./src/mock",
      outputPath: "./src/mock/result",
    };
    const plugin = new AssetsToMap(config);

    // when
    let result = plugin.writeFile();

    let data = fs.readFileSync("./src/mock/result/final.json").toString();

    // then
    expect(result).toBeTruthy();
    expect(data).toMatchSnapshot();
  });

  describe("single file provided", () => {
    test("writeFiles format set to default", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
        },
        filename: "final",
        outputPath: "./src/mock/result",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to json", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "json",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to yaml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "yaml",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.yaml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to toml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "toml",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.toml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });
  });

  describe("multiple files provided", () => {
    test("writeFiles format set to default", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        outputPath: "./src/mock/result",
        filename: "final",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to json", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "json",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to yaml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "yaml",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.yaml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to toml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "toml",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.toml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });
  });

  describe("remove files inlined", () => {
    test("writeFiles format set to default", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        outputPath: "./src/mock/result",
        filename: "final",
        deleteFiles: true,
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to json", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "json",
        deleteFiles: true,
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.json").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to yaml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "yaml",
        deleteFiles: true,
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.yaml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });

    test("writeFiles format set to toml", () => {
      // with
      const config = {
        files: {
          success: "./src/mock/mock.js",
          style: "./src/mock/mock2.css",
        },
        filename: "final",
        outputPath: "./src/mock/result",
        format: "toml",
      };
      const plugin = new AssetsToMap(config);

      // when
      let result = plugin.writeFile();

      let data = fs.readFileSync("./src/mock/result/final.toml").toString();

      // then
      expect(result).toBeTruthy();
      expect(data).toMatchSnapshot();
    });
  });
});
