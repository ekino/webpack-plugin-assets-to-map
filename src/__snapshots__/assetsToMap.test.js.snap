// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsToMap multiple files provided writeFiles format set to default 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;

exports[`AssetsToMap multiple files provided writeFiles format set to json 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;

exports[`AssetsToMap multiple files provided writeFiles format set to toml 1`] = `
"success = '''
                console.log(\\"mock.js file\\");'''
                
style = '''
                .mock2{
    content: \\"mock2.css\\";
    background: rebeccapurple;
    color: aliceblue;
}'''
                
"
`;

exports[`AssetsToMap multiple files provided writeFiles format set to yaml 1`] = `
"success : |
                console.log(\\"mock.js file\\");'''
                
style : |
                .mock2{     content: \\"mock2.css\\";     background: rebeccapurple;     color: aliceblue; }'''
                
"
`;

exports[`AssetsToMap remove files inlined writeFiles format set to default 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;

exports[`AssetsToMap remove files inlined writeFiles format set to json 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;

exports[`AssetsToMap remove files inlined writeFiles format set to toml 1`] = `
"success = '''
                console.log(\\"mock.js file\\");'''
                
style = '''
                .mock2{
    content: \\"mock2.css\\";
    background: rebeccapurple;
    color: aliceblue;
}'''
                
"
`;

exports[`AssetsToMap remove files inlined writeFiles format set to yaml 1`] = `
"success : |
                console.log(\\"mock.js file\\");'''
                
style : |
                .mock2{     content: \\"mock2.css\\";     background: rebeccapurple;     color: aliceblue; }'''
                
"
`;

exports[`AssetsToMap single file provided writeFiles format set to default 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\"}"`;

exports[`AssetsToMap single file provided writeFiles format set to json 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\"}"`;

exports[`AssetsToMap single file provided writeFiles format set to toml 1`] = `
"success = '''
                console.log(\\"mock.js file\\");'''
                
"
`;

exports[`AssetsToMap single file provided writeFiles format set to yaml 1`] = `
"success : |
                console.log(\\"mock.js file\\");'''
                
"
`;

exports[`AssetsToMap writeFiles sourcePath and outputPath should be okay with a trailing slash 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;

exports[`AssetsToMap writeFiles sourcePath and outputPath should be okay without a trailing slash 1`] = `"{\\"success\\":\\"console.log(\\\\\\"mock.js file\\\\\\");\\",\\"style\\":\\".mock2{\\\\n    content: \\\\\\"mock2.css\\\\\\";\\\\n    background: rebeccapurple;\\\\n    color: aliceblue;\\\\n}\\"}"`;
